cgo/sync.go

```diff
@@ -42,6 +43,10 @@ func syncWallet(cName, cPeers *C.char) *C.char {
 		},
 		FetchMissingCFiltersStarted: func() {
 			w.syncStatusMtx.Lock()
+			if w.rescanning {
+				w.syncStatusMtx.Unlock()
+				return
+			}
```

Comment:
It looks like these are being called during the rescan, which is strange. This may be why the status doesnt say synced without the bandaid.

Reply:
> We probably don't want to allow rescanning when sync is on.

Do you mean stop sync then restart it? Is that what other wallets do?

Reply:
Oh sorry missed this. Now checking if synced before rescan https://github.com/decred/libwallet/compare/f2eca88e6d25a98ab03ee087bf65b69bed3fe73c..1f2c9ca6c4eb07607509400459b8e1164e6fe83b

Not sure why I was holding the wallet lock. Looks ok without the lock?